autopush_rs/src/util/megaphone.rs

```diff
@@ -60,12 +60,35 @@ struct ServiceRevision {
 
 // A provided Service/Version used for `ChangeList` initialization, client comparisons, and
 // outgoing deltas
-#[derive(Clone, Debug, PartialEq)]
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
 pub struct Service {
     service_id: String,
     version: String,
 }
 
+// Handy From impls for common hashmap to/from conversions
+impl From<(String, String)> for Service {
+    fn from(val: (String, String)) -> Service {
+        Service { service_id: val.0, version: val.1 }
+    }
+}
+
+impl From<Service> for (String, String) {
+    fn from(svc: Service) -> (String, String) {
+        (svc.service_id, svc.version)
+    }
+}
+
+impl Service {
+    pub fn from_hashmap(val: HashMap<String, String>) -> Vec<Service> {
+        val.into_iter().map(|v| v.into()).collect()
+    }
+
+    pub fn to_hashmap(service_vec: Vec<Service>) -> HashMap<String, String> {
```

Comment:
this function consumes service_vec so one of the rust naming conventions prefers it called into_hashmap https://deterministic.space/elegant-apis-in-rust.html#more-method-name-conventions